packages/blockchain/src/blockchain.ts

```diff
@@ -452,6 +466,8 @@ export class Blockchain implements BlockchainInterface {
         await this.dbManager.batch(ops)
 
         await this.consensus?.newBlock(block, commonAncestor, ancestorHeaders)
+        this.DEBUG &&
+          this._debug(`put block: number ${block.header.number} hash ${bytesToHex(block.hash())}`)
```

Comment:
Do not have the full picture but if the hash is available beforehand it would be good to not calculate (call) again for performance reasons but simply take an eventually existing assigned variable.

Can you have a look through the `hash()` usages (several below as well) where possible?

Reply:
Fixed